src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValueNotTrimmable.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Text.Json.Nodes
+{
+    /// <summary>
+    /// Not trim-safe since it calls JsonSerializer.Serialize(JsonSerializerOptions).
+    /// </summary>
+    internal sealed partial class JsonValueNotTrimmable<TValue> : JsonValue<TValue>
+    {
+        public JsonValueNotTrimmable(TValue value, JsonNodeOptions? options = null) : base(value, options) { }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "The methods used to create this JsonValue are marked RequiresUnreferencedCode.")]
```

Comment:
This isn't true. The ctor of this class needs to be marked as `RequiresUnreferencedCode`.

I believe that will catch a bug in the `JsonObject.Dynamic.cs` file where we are creating a `JsonValueNotTrimmable`, but aren't marking that as `RequiresUnreferencedCode`.

Reply:
Thanks, I'll address this and other feedback in a new PR.